client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/site-preview/index.tsx

```diff
@@ -1,19 +1,43 @@
+import { SiteThumbnail, Spinner } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
-import React, { FC } from 'react';
+import { FC, useEffect, useRef } from 'react';
 import { useQuery } from 'calypso/landing/stepper/hooks/use-query';
+import { useSitePreviewMShotImageHandler } from './hooks/use-site-preview-mshot-image-handler';
 import './style.scss';
 
 export const SitePreview: FC = () => {
 	const translate = useTranslate();
 	const fromUrl = useQuery().get( 'from' ) || '';
 
+	const previewRef = useRef< HTMLDivElement >( null );
+	const { mShotsOption, updateDimensions } = useSitePreviewMShotImageHandler();
+
+	useEffect( () => {
+		updateDimensions( previewRef );
+
+		// Throttle the resize event handling.
+		let resizeTimeout: ReturnType< typeof setTimeout >;
```

Comment:
WDYT to use lodash here to simplify a bit?

Reply:
Makes sense, updated here https://github.com/Automattic/wp-calypso/pull/92355/commits/7581f8398d81bfa3342388b61bd81799db6411e2. Thanks!